storybookjs/storybook
# Vite: Add partial SvelteKit support

Issue: #19280

This PR is a first stab at setting up a SvelteKit repro and getting it to work out of the box.

## TODO

- [x] Test that base `vite-svelte` framework still works
- [x] Test that TS SvelteKit framework works.
- [x] Test that it works when published to verdaccio

## What I did

- Add repros via [`create-svelte-with-args`](https://github.com/storybookjs/create-svelte-with-args)
- Removed the `sveltePreprocess` option from the template as that was causing ESM vs CJS issues and we decided it was an anti-pattern anyway since they should automatically be picked up by the Vite config
- Remove `vite-plugin-svelte-kit` whenever it is recognised in a Vite config because it conflicts with Storybook's VIte server handling. This has some drawbacks that certain SvelteKit features aren't available in Storybook but for now we consider that a general issue that needs to be solved in SvelteKit.

## How to test

1. `git checkout origin/jeppe/sb-544-sveltekit`
2. Use Node 16 instead of Node 14 (SvelteKit can't be built using v14) - eg. `nvm use 16`
3. Generate repro with `cd code && yarn generate-repros-next --template svelte-kit/skeleton-js`
4. `cd ../sandbox/svelte-kit-skeleton-js && yarn install`
5. `yarn storybook`
6. See that all basic stories work.
7. Repeat for SvelteKit with TypeScript: `svelte-kit/skeleton-ts`

## Out of scope

I want to include a few stories based on the Svelte CSF format in the sandbox so CI also ensures that they work. That's another PR.
